test/lib/commands/publish.js

```diff
@@ -49,6 +52,7 @@ t.test('respects publishConfig.registry, runs appropriate scripts', async t => {
   t.equal(fs.existsSync(path.join(prefix, 'scripts-prepublish')), false, 'did not run prepublish')
   t.equal(fs.existsSync(path.join(prefix, 'scripts-publish')), true, 'ran publish')
   t.equal(fs.existsSync(path.join(prefix, 'scripts-postpublish')), true, 'ran postpublish')
+  t.same(logs.warn, ['Unknown publishConfig config "other". This will stop working in the next major version of npm.'])
```

Comment:
is this accurate? i thought the intention was to leave them as warnings for the foreseeable future.

Reply:
Yes it's accurate.  The warning is for the next semver major npm, at which time they will be ignored and not in npm config nor the flatOptions it sends to modules nor part of env exports.  There will always be a warning that they are being ignored.

This is in contrast to all config NOT in publishConfig which will result in an exit of npm with an error.

Reply:
but `other`, specifically, doesn't work now - it's ignored now, and will still be ignored in the future, and will be warned forever. should there be two distinct messages, one for things that are actually unknown, vs another for things that are known but planned to be ignored in the future?

Reply:
> it's ignored now

It is not ignored.

```sh
$ npm pkg get publishConfig
{
  "foo": "bar"
}
$ npm config ls
...
; "publishConfig" from /Users/wraithgar/Development/scratch/publish/package.json
; This set of config values will be used at publish-time.

foo = "bar"
```

They are only "ignored" in the sense that npm currently doesn't look for them.  They're set though.  If we ever added them as real config items, folks who have them set would have whatever they have set used for the values of those configs, very unexpectedly.

> and will still be ignored in the future

How can you know this?

How would you programmatically go about guessing which things in a user's config are something they intended npm to recognize?

Reply:
Ah, I think that clarifies - in other words, "work" here means "ends up in the npm config and env vars", which even if npm doesn't look at them at runtime, will be available to user scripts. However, after the next major, they will not end up in the npm config at all - npm will ignore them completely.

I think the confusion for me was "working" - to me publishConfig is just a bucket, I don't think of it primarily as a way to seed npm's config and env vars.